specs/AnimatedVisualPlayer Spec.md

```diff
@@ -0,0 +1,137 @@
+AnimatedVisualPlayer Spec
+===
+
+# Background
+
+Today, developers can create animated visuals to use in their UI by using the AnimatedVisualPlayer control. 
+The AnimatedVisualPlayer hosts and controls playback of an animated Visual tree, integrating custom motion graphic content with XAML UI. 
+For instance, the AnimatedVisualPlayer is used to display and control Lottie animations.
+
+Before a developer can use an animated visual in their code base, they must convert it to code using the LottieGen tool. 
+LottieGen is a tool for generating C#, C++, and other outputs from Lottie / Bodymovin JSON files. 
+LottieGen is built as a .NET global tool, which means it will run anywhere .NET Core is available, including Linux and Mac.
+
+Once the visual’s code has been generated using LottieGen, the developer can use the visual in their code base. 
+
+There are some performance issues with the current implementation of LottieGen and AnimatedVisual Player: 
+1)	Complex Lottie animations can instantiate a lot of composition animations which may impact performance and overall battery life. 
+2)	Lottie files that have a lot of complex animations even when they are paused, have some cost to performance. 
+The Idle (paused) lottie animations use the same amount of CPU resources as when they are playing the actual animation.
+
+We would like to update AnimatedVisualPlayer, AnimatedIcon, and the interface generated by LottieGen to help reduce the number of composition 
+animations created when the animation is idle or paused to help improve performance when a developer uses Lottie Animations with AnimatedVisualPlayer. 
+
+This spec will focus on changes for LottieGen and AnimatedVisualPlayer. 
+
+This will require 3 changes: 
+1.	Create a new interface for which an implementation is generated by LottieGen called IAnimatedVisual2, that can destroy and instantiate animations.
+2.	Create a new interface IAnimatedVisualSource3 that should allow instantiate visuals without animations.
+3.	Add a CacheMode property to AnimatedVisualPlayer that will control when animations should be destroyed and when they should be instantiated.
+
+
+# API Pages
+
+## AnimatedVisualPlayer.CacheMode property
+
+CacheMode object that can be set to 'Always' or 'None' to create/destroy animations when the PlayAsync method of AnimatedVisualPlayer is complete. 
+
+Defaults to 'Always'.
+
+The two modes for the property will specify the behavior the AnimatedVisualPlayer will have when the player is idle (when PlayAsync is complete).
+
+|Mode| Behavior|
+|---- | -------|
+|Always| AnimatedVisualPlayer will instantiate animations and not destroy any when player is idle. |
```

Comment:
Would this cause animations to be instantiated on load ? Also, when this property is set, it would immediate instantiate ?

Reply:
Yes, this would cause animations to be instantiated on load and behave how AnimatedVisualPlayer does today. The latter is a great question and I believe it would instantiate when the property is set. @AndrewB330 what are your thoughts?

Reply:
@ranjeshj @MarissaMatt Yes, when we set the property to Always and there is nothing in play, we will instantiate all animations immediately.